unity-renderer/Assets/DCLPlugins/ECS7/InternalECSComponents/Interfaces/Extensions/InternalInputEventResultsExtensions.cs

```diff
@@ -11,7 +11,7 @@ public static class InternalInputEventResultsExtensions
         {
             var entityId = data.hit?.EntityId?? SpecialEntityId.SCENE_ROOT_ENTITY;
 
-            var model = component.GetFor(scene, entityId)?.model ?? new InternalInputEventResults();
+            var model = component.GetFor(scene, entityId)?.model ?? new InternalInputEventResults(new List<InternalInputEventResults.EventData>(20));
```

Comment:
here as well - is it possible to cache, clear and re-use instead of creating new instances?